x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/IndicesPermission.java

```diff
@@ -530,7 +520,90 @@ public IndicesAccessControl authorize(
                 )
             );
         }
-        return new IndicesAccessControl(overallGranted, unmodifiableMap(indexPermissions));
+        return unmodifiableMap(indexPermissions);
+    }
+
+    /**
+     * Returns {@code true} if action is granted for all {@code requestedResources}.
+     * If action is not granted for at least one resource, this method will return {@code false}.
+     */
+    private boolean isActionGranted(final String action, final Map<String, IndexResource> requestedResources) {
+
+        if (requestedResources.isEmpty()) {
+            return true;
+        }
```

Comment:
Please just use an assert here (since it is never the case). It is slightly distressing that an authz method `isActionGranted` has a special case for the granted case (ie it would be less distressing if the return was `false`).

Reply:
Added assert in [0a3dd7d](https://github.com/elastic/elasticsearch/pull/88662/commits/0a3dd7d05332ea431873cda303d04c5d00db9f91)